app/src/main/java/com/onewelcome/showcaseapp/ui/screens/sections/SdkInitializationScreen.kt

```diff
@@ -2,52 +2,154 @@ package com.onewelcome.showcaseapp.ui.screens.sections
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
 import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.Surface
+import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavController
 import com.onewelcome.showcaseapp.R
+import com.onewelcome.showcaseapp.ui.components.ExpandableCard
+import com.onewelcome.showcaseapp.ui.components.NumberSettingTextField
+import com.onewelcome.showcaseapp.ui.components.SettingCheckbox
+import com.onewelcome.showcaseapp.ui.theme.Dimensions
 import com.onewelcome.showcaseapp.viewmodel.SdkInitializationViewModel
+import com.onewelcome.showcaseapp.viewmodel.SdkInitializationViewModel.State
+import com.onewelcome.showcaseapp.viewmodel.SdkInitializationViewModel.UiEvent
 
 @Composable
 fun SdkInitializationScreen(
   navController: NavController,
   viewModel: SdkInitializationViewModel = hiltViewModel()
 ) {
-  SdkInitializationScreenContent(onInitializeSdkClicked = { viewModel.initializeOneginiSdk() })
+  SdkInitializationScreenContent(
+    uiState = viewModel.uiState,
+    onNavigateBack = { navController.popBackStack() },
+    onEvent = { viewModel.onEvent(it) })
 }
 
 @Composable
-private fun SdkInitializationScreenContent(onInitializeSdkClicked: () -> Unit) {
-  Surface(
-    modifier = Modifier.fillMaxSize(),
-    color = MaterialTheme.colorScheme.background
-  ) {
+private fun SdkInitializationScreenContent(
+  uiState: State,
+  onNavigateBack: () -> Unit,
+  onEvent: (UiEvent) -> Unit
+) {
+  Scaffold(
+    topBar = { TopBar(onNavigateBack) },
+  ) { innerPadding ->
     Column(
-      verticalArrangement = Arrangement.Center,
-      horizontalAlignment = Alignment.CenterHorizontally
+      modifier = Modifier
+        .padding(innerPadding)
+        .padding(start = Dimensions.standardPadding, end = Dimensions.standardPadding)
     ) {
+      SettingsSection(
+        modifier =
+          Modifier
+            .weight(1f)
+            .padding(bottom = Dimensions.smallPadding),
+        uiState = uiState,
+        onEvent = onEvent
+      )
       Button(
-        modifier = Modifier.wrapContentSize(),
-        onClick = { onInitializeSdkClicked.invoke() }
+        modifier = Modifier.fillMaxWidth(),
+        onClick = { onEvent(UiEvent.InitializeOneginiSdk) }
       ) {
         Text(stringResource(R.string.button_initialize_sdk))
       }
     }
   }
 }
 
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun TopBar(onNavigateBack: () -> Unit) {
```

Comment:
I think we could extract this in similar way as `NumberSettingTextField` and `SettingsCheckbox` because `TopBar` will be used in many places. 

I'm pretty surprised with the need of stating
`windowInsets = WindowInsets(0.dp),`
by default `TopAppBar` has enormous windows inset? It is suspicious 🤔 Where does it come from?

Reply:
As you already noticed it got extracted in the next PR 😄 
Initially I thought it has something to do with edge to edge setting as that was mentioned by people on the Internet. But when I look at it now I think those insets get doubled because we have nested Scaffolds - one in the BottomNavigationBar and here is the second one. I'm not sure if there's a better solution to that 🤔